ipq806x: provide wifi mac from dts

Ansuel opened the pull request:
We have a dedicated patch to patch ath10k mac address. Directly use the dts to provide the mac instead of use a dedicated script.

@adschm What do you think about this? Also i wonder if we can generalize the wifi bridge (as every ipq8064 board use a qca9980 and every ipq8065 board use a qca9984 chip) and use a tag to set the mac address

adschm commented on target/linux/generic/pending-5.4/681-NET-add-of_get_mac_address_mtd.patch:
```diff
@@ -79,7 +79,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 +		return NULL;
 +
-+	if (!of_property_read_u32(np, "mtd-mac-address-increment", &mac_inc))
++	if (!of_property_read_s32(np, "mtd-mac-address-increment", &mac_inc))
```
Why has this been working already on ath79?

Ansuel replied:
Don't really know... I just notice that in some ath79 dts we have negative value but looking at the code isn't this wrong?  
Since mac_inc is u32 and proprety_read treat the value as unsigned i can't really find where the negative value is converted to unsigned... Could be that nobody notice that?

adschm replied:
Since I put some effort into having MAC addresses correct, I have discussed so many cases in PRs that I can't imagine negative increments not working on ath79.

Ansuel replied:
Actually.... I can make a quick test with my router and check if the value is decremented correctly

adschm commented:
What happens if the caldata contains a valid dummy MAC address? Will the value from DTS be overwritten or will the DTS value win anyway?

Ansuel commented:
> What happens if the caldata contains a valid dummy MAC address? Will the value from DTS be overwritten or will the DTS value win anyway?

From the patch description
`Devices may provide their own MAC address via system firmware (e.g.,
device tree), especially in the case where the device doesn't have a
useful EEPROM on which to store its MAC address (e.g., for integrated
Wifi).

Use the generic device helper to retrieve the MAC address, and (if
present) honor it above the MAC address advertised by the card.`

Ansuel commented:
@adschm ok i made some test... the negative value works are expected... 

Another problem arise... I'm still searching the reason (I think it's due to the kernel changes) but the "mtd-mac-address" property doesn't work anymore... 

With early discovery, to fix this a patch for the ath10k driver is needed. Do you think this could be acceptable or it's better to leave the mac patch with the hotplug script?

CodeFetch commented:
@Ansuel Please try to upstream the changes to ath10k first.

Ansuel committed:
ipq806x: provide mac-address from dts

Use nvmem framework for supported mac-address stored
in nvmem cells and drop mac patch function for hotplug
script for supported devices.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel commented:
@adschm can you check this? Now that we have all the nvmem fixes this can be implemented right away.
Tested this on r7800

Ansuel commented:
@adschm sorry for the spam... just trying to clear old pr

mans0n commented on target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8064-ad7200-c2600.dtsi:
```diff
@@ -249,11 +249,42 @@
+		ranges;
+
+		wifi@1,0 {
+			compatible = "pci168c,0040";
```
Shouldn't we use `"qcom,ath10k"` here?

adschm replied:
@Ansuel Ping. qcom,ath10k instead?

Ansuel replied:
I never tested the ath10k compatible. I put this as it's an exact description of the wifi card in the device

adschm replied:
Let's use it for now then.

mans0n commented on target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:
```diff
@@ -20,7 +20,6 @@ case "$FIRMWARE" in
 		;;
 	buffalo,wxr-2533dhp)
 		caldata_extract "ART" 0x1000 0x2f20
-		ath10k_patch_mac $(mtd_get_mac_binary ART 0x1e)
```
Without `ath10k_patch_mac` we can now consolidate duplicate cases.

Ansuel replied:
Can't understand what do you mean here.

adschm replied:
We have lots of block with identical setup, and those can be merged. I've already done that in a separate patch.

adschm commented:
@Ansuel My problem is that I like the idea, but I cannot judge whether the implementation is correct.

adschm commented:
> @Ansuel My problem is that I like the idea, but I cannot judge whether the implementation is correct.

Or, to make it a little bit more constructive: I'm not familiar with this bridge@ setup/syntax. If somebody will review or ack that, and if we get test reports from two or three devices that the MAC addresses are actually there, I'd be happy to merge this.

Ansuel commented:
Greearb accepted the nvmem patch and also this is just a continue of chukeey work.
Only problem is with ath10k that can be easily patched. (Greearb is patched from version 5.13)

Ansuel commented:
Anyway this is tested on7800 and I checked wikidevi to set the correct
bridge Id (hoping wikidevi is not wrong)

Il Dom 3 Ott 2021, 14:24 Adrian Schmutzler ***@***.***> ha
scritto:

> @Ansuel <https://github.com/Ansuel> My problem is that I like the idea,
> but I cannot judge whether the implementation is correct.
>
> Or, to make it a little bit more constructive: I'm not familiar with this
> bridge@ setup/syntax. If somebody will review or ack that, and if we get
> test reports from two or three devices that the MAC addresses are actually
> there, I'd be happy to merge this.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/3403#issuecomment-932940411>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQQWN4BHPPXZ4J63FDTUFBDPJANCNFSM4RIOEYYQ>
> .
>

adschm commented:
> Greearb accepted the nvmem patch and also this is just a continue of chukeey work. Only problem is with ath10k that can be easily patched. (Greearb is patched from version 5.13)

So, this breaks with ath10k non-CT? I don't know, are there people using non-CT ath10k on this platform? I'm aware that many do on other targets like ath79 ...

Ansuel commented:
The patch mac will still work. The only breakage is if someone use
unpatched ath10k with this implementation
(So ipq806x with ath10k non ct)
The breakage is random Mac

Il Dom 3 Ott 2021, 14:30 Adrian Schmutzler ***@***.***> ha
scritto:

> Greearb accepted the nvmem patch and also this is just a continue of
> chukeey work. Only problem is with ath10k that can be easily patched.
> (Greearb is patched from version 5.13)
>
> So, this breaks with ath10k non-CT? I don't know, are there people using
> non-CT ath10k on this platform? I'm aware that many do on other targets
> like ath79 ...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/3403#issuecomment-932941297>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQVZYW72OC6MVZPDZ33UFBEGFANCNFSM4RIOEYYQ>
> .
>

castiel652 commented:
> > Greearb accepted the nvmem patch and also this is just a continue of chukeey work. Only problem is with ath10k that can be easily patched. (Greearb is patched from version 5.13)
> 
> So, this breaks with ath10k non-CT? I don't know, are there people using non-CT ath10k on this platform? I'm aware that many do on other targets like ath79 ...

I am using the non-CT for my R7800 and I know there's a small group of people on the OpenWrt forum using it.

Ansuel commented:
Then we just patch ath10k... it's really a one line change...

Il Dom 3 Ott 2021, 16:03 Zhijun You ***@***.***> ha scritto:

> Greearb accepted the nvmem patch and also this is just a continue of
> chukeey work. Only problem is with ath10k that can be easily patched.
> (Greearb is patched from version 5.13)
>
> So, this breaks with ath10k non-CT? I don't know, are there people using
> non-CT ath10k on this platform? I'm aware that many do on other targets
> like ath79 ...
>
> I am using the non-CT for my R7800 and I know there's a small group of
> people on the OpenWrt forum using it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/3403#issuecomment-932957879>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQXK7WKNHHBWAUCGKGTUFBPCFANCNFSM4RIOEYYQ>
> .
>

adschm commented:
> Then we just patch ath10k... it's really a one line change... Il Dom 3 Ott 2021, 16:03 Zhijun You ***@***.***> ha scritto:

Yes, please. I don't see why we should merge this when it effectively reduces the feature-set.

If a corresponding change has already been accepted for CT, please reference that in the patch for non-CT, so it's easier to review it.

Ansuel commented:
@adschm will refresh this pr with the ath10k patch. Hope it will be good.

Ansuel commented:
@adschm i just notice when we introduced the nvmem changes we already patched ath10k to support this.
So this can be merged right away with no regression

adschm commented on target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8064-d7800.dts:
```diff
@@ -161,13 +161,45 @@
+	};
 };
 
 &usb3_1 {
```
@Ansuel Are these two attached to usb nodes by intention?!

Ansuel replied:
not at all... i assume a rebase problem

adschm replied:
I've already pulled it to my local tree, so no need to rebase for you. Just wanted to know. I'll fix it here.
